jwst/outlier_detection/outlier_detection_tso.py

```diff
@@ -0,0 +1,211 @@
+        median_model = dm.CubeModel(data=medians)
+
+        # Save median model if pars['save_intermediate_results'] is True
+        # this will be a CubeModel with rolling median values.
+        if self.outlierpars['save_intermediate_results']:
+            with dm.open(weighted_cube[0]) as dm0:
+                median_model.update(dm0)
+            self.save_median(median_model) 
+
+        # no need for blotting, resample is turned off for TSO
+        # go straight to outlier detection
+        log.info("Flagging outliers")
+        flag_cr(self.input_models, median_model, **self.outlierpars)
+
+        if self.converted:
+            # Make sure actual input gets updated with new results
+            for i in range(len(self.inputs)):
+                self.inputs[i].dq = self.input_models.dq[i]
+    
+    
+    def weight_no_resample(self):
+        """
+        give weights to model without resampling
+
+        Notes
+        -----
+        Prior to PR #8473, the `build_driz_weight` function was used to
+        create the weights for the input models for TSO data. However, that
+        function was simply returning a copy of the DQ array because the 
+        var_noise was not being passed in by calwebb_tso3. As of PR #8473,
+        a cube model that includes the var_noise is passed into TSO 
+        outlier detection, so `build_driz_weight` would weight the cube model
+        by the variance. Therefore `build_driz_weight` was removed in order to
+        preserve the original behavior. If it is determined later that exposure
+        time or inverse variance weighting should be used here, build_driz_weight
+        should be re-implemented.
+        """
+        weighted_cube = self.input_models.copy()
+        dqmask = build_mask(self.input_models.dq, self.outlierpars['good_bits'])
+        weighted_cube.wht = dqmask.astype(np.float32)
+        return weighted_cube
+    
+
+    def compute_rolling_median(self, model: dm.CubeModel, weight_threshold: np.ndarray, w:int = 25) -> np.ndarray:
+        '''
+        Set bad and low-weight data to NaN, then compute the rolling median over the time axis.
+
+        Parameters
+        ----------
+        model : ~jwst.datamodels.CubeModel
+            The input cube model
+
+        weight_threshold : np.ndarray
+            The weight thresholds for each integration.
+
+        w : int
+            The window size for the rolling median. 
+
+        Returns
+        -------
+        np.ndarray
+            The rolling median of the input data. Same dimensions as input.
+        '''
+
+        sci = model.data
+        weight = model.wht
+        badmask = np.less(weight, weight_threshold)
+        log.debug("Percentage of pixels with low weight: {}".format(
+                np.sum(badmask) / weight.size * 100))
+
+        # Fill resampled_sci array with nan's where mask values are True
+        for f1, f2 in zip(sci, badmask):
```

Comment:
Is it necessary to loop over the individual planes? Could one just have `sci[badmask] = np.nan`?